qemu/tests/qemu_guest_agent.py

```diff
@@ -1580,14 +1580,32 @@ def gagent_check_file_read(self, test, params, env):
         # if guest os resource is enough, will return no error.
         # else it will return error like "insufficient system resource"
         # which is expected
-        try:
-            self.gagent.guest_file_read(ret_handle, count=1000000000)
-        except guest_agent.VAgentCmdError as detail:
-            info_insuffi = "Insufficient system resources exist to"
-            info_insuffi += " complete the requested service"
-            if not re.search(info_insuffi, str(detail)):
-                test.fail("Return error but is not the desired information: "
-                          "('%s')" % str(detail))
+        if params.get("os_type") == "linux":
+            logging.info("Get guest agent's main version for linux guest.")
+            cmd_qemu_ver = params["cmd_get_qemu_version"]
+            qemu_ver = int(session.cmd_output(cmd_qemu_ver).strip())
```

Comment:
Please call `utils_qemu.get_qemu_version` to get qemu version.

Reply:
Hi @YongxueHong ,I tried to use `/bin/qemu-ga --version` to check qemu version and the result is ' QEMU Guest Agent 5.1.0 ' ,but I found this match statement from utils_qemu.py `QEMU_VERSION_RE = re.compile(r"QEMU (?:PC )?emulator version\s"
                             r"([0-9]+\.[0-9]+\.[0-9]+)"
                             r"(?:\s\((.*?)\))?")`
and I cannot match this type of output,can you please give me any advice? thanks

Reply:
> Hi @YongxueHong ,I tried to use `/bin/qemu-ga --version` to check qemu version and the result is ' QEMU Guest Agent 5.1.0 ' ,but I found this match statement from utils_qemu.py `QEMU_VERSION_RE = re.compile(r"QEMU (?:PC )?emulator version\s" r"([0-9]+\.[0-9]+\.[0-9]+)" r"(?:\s\((.*?)\))?")`
> and I cannot match this type of output,can you please give me any advice? thanks

From your comment: `# Higher than qemu-5.0(rhel8.3.0), the count number is changed,` so what is your checkpoint? the qemu-kvm version or the qemu-agent version?

Reply:
If you wanna get the qemu-agent version, please refer to line 1543~1546 which is to get the main version of guest agent.
```
logging.info("Get guest agent's main version for linux guest.")
qga_ver = session.cmd_output(params["gagent_pkg_check_cmd"])
ver_main = int(re.findall(pattern, qga_ver)[0])
pattern = r"guest-agent-(\d+).\d+.\d+-\d+"
```
Please wrap the above routines into a function so that could call it in the future if there is also the same requirement.

Reply:
> > Hi @YongxueHong ,I tried to use `/bin/qemu-ga --version` to check qemu version and the result is ' QEMU Guest Agent 5.1.0 ' ,but I found this match statement from utils_qemu.py `QEMU_VERSION_RE = re.compile(r"QEMU (?:PC )?emulator version\s" r"([0-9]+\.[0-9]+\.[0-9]+)" r"(?:\s\((.*?)\))?")`
> > and I cannot match this type of output,can you please give me any advice? thanks
> 
> From your comment: `# Higher than qemu-5.0(rhel8.3.0), the count number is changed,` so what is your checkpoint? the qemu-kvm version or the qemu-agent version?

There is no command `qemu-kvm ` so that I just check qemu version with command `qemu-ga --version`,or maybe I got something missed ?

Reply:
> If you wanna get the qemu-agent version, please refer to line 1543~1546 which is to get the main version of guest agent.
> 
> ```
> logging.info("Get guest agent's main version for linux guest.")
> qga_ver = session.cmd_output(params["gagent_pkg_check_cmd"])
> ver_main = int(re.findall(pattern, qga_ver)[0])
> pattern = r"guest-agent-(\d+).\d+.\d+-\d+"
> ```
> 
> Please wrap the above routines into a function so that could call it in the future if there is also the same requirement.

I'd like to create a function with this,thanks.

Reply:
From the codes of `gagent_check_file_read`, I guess you wanna get the main version of qemu-guest-agent.
So better to show clear, like line 1562~1563 to tell others which factor decides the count value.
```
            # for windows os or qga version > 2 for linux os,
            # the large count number is an invalid parameter from qga.
```
since your comments are not clear.
`Higher than qemu-5.0(rhel8.3.0), the count number is changed,` others don't know what is the version `qemu-5.0`.
So better to update your comments, not create ambiguity.

Reply:
okay,you're right,I've changed that comments already.